sklearnex/linear_model/incremental_linear.py

```diff
@@ -0,0 +1,369 @@
+                copy=self.copy_X,
+                force_all_finite=False,
+            )
+
+        assert hasattr(self, "_onedal_estimator")
+        if self._need_to_finalize:
+            self._onedal_finalize_fit()
+        X = check_array(X, dtype=[np.float64, np.float32])
+        return self._onedal_estimator.predict(X, queue)
+
+    def _onedal_partial_fit(self, X, y, queue=None):
+        first_pass = not hasattr(self, "n_samples_seen_") or self.n_samples_seen_ == 0
+
+        if sklearn_check_version("1.2"):
+            self._validate_params()
+
+        if sklearn_check_version("1.0"):
+            X, y = self._validate_data(
+                X,
+                y,
+                dtype=[np.float64, np.float32],
+                reset=first_pass,
+                copy=self.copy_X,
+                multi_output=True,
+            )
+        else:
+            X = check_array(
+                X,
+                dtype=[np.float64, np.float32],
+                copy=self.copy_X,
+            )
+            y = check_array(
+                y,
+                dtype=[np.float64, np.float32],
+                copy=False,
+                ensure_2d=False,
+            )
+
+        if first_pass:
+            self.n_samples_seen_ = X.shape[0]
+            self.n_features_in_ = X.shape[1]
+        else:
+            self.n_samples_seen_ += X.shape[0]
+        onedal_params = {"fit_intercept": self.fit_intercept, "copy_X": self.copy_X}
+        if not hasattr(self, "_onedal_estimator"):
+            self._onedal_estimator = self._onedal_incremental_linear(**onedal_params)
+        self._onedal_estimator.partial_fit(X, y, queue)
+        self._need_to_finalize = True
+
+    def _onedal_finalize_fit(self):
+        assert hasattr(self, "_onedal_estimator")
+        is_good_for_onedal = self.n_samples_seen_ >= self.n_features_in_ + int(
+            self.fit_intercept
+        )
+        if not is_good_for_onedal:
+            raise ValueError("Not enough samples to finalize")
+        self._onedal_estimator.finalize_fit()
+        self._need_to_finalize = False
+
+    def _onedal_fit(self, X, y, queue=None):
+        if sklearn_check_version("1.0"):
+            X, y = self._validate_data(
+                X, y, dtype=[np.float64, np.float32], copy=self.copy_X, multi_output=True
+            )
+        else:
+            X = check_array(
+                X,
+                dtype=[np.float64, np.float32],
+                copy=self.copy_X,
+            )
+            y = check_array(
+                y,
+                dtype=[np.float64, np.float32],
+                copy=False,
+                ensure_2d=False,
+            )
+
+        n_samples, n_features = X.shape
+        if self.batch_size is None:
+            self.batch_size_ = 5 * n_features
+        else:
+            self.batch_size_ = self.batch_size
+
+        self.n_samples_seen_ = 0
+        if hasattr(self, "_onedal_estimator"):
+            self._onedal_estimator._reset()
```

Comment:
how well does this wipe the memory on oneDAL side? does it start from scratch?

Reply:
I didn't think about it. I guess python garbage collector should resolve it somehow. this method I just creates new empty partial_result and put it instead the existing to class attribute